ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+                nMMLanguage = MM_SPA_LANGUAGE;
+            else
+                nMMLanguage = MM_ENG_LANGUAGE;
+        }
+        else
+            nMMLanguage = MM_DEF_LANGUAGE;  // Default
+
+        /* ---------------------------------------------------------------- */
+        /*      Preparing to write the layer                                */
+        /* ---------------------------------------------------------------- */
+        if (!STARTS_WITH(pszFilename, "/vsistdout"))
+        {
+            // Init the feature (memory, num,...)
+            if (MMInitFeature(&hMMFeature))
+            {
+                bValidFile = false;
+                return;
+            }
+
+            // Init the Layers (not in disk, only in memory until
+            // the first element is readed)
+            CPLDebug("MiraMon", "Initializing MiraMon points layer...");
+            if (MMInitLayer(&hMiraMonLayerPNT, pszFilename, nMMVersion,
+                            nMMRecode, nMMLanguage, nMMMemoryRatio, nullptr,
+                            MM_WRITTING_MODE, MMMap))
+            {
+                bValidFile = false;
+                return;
+            }
+            hMiraMonLayerPNT.bIsBeenInit = 0;
+
+            CPLDebug("MiraMon", "Initializing MiraMon arcs layer...");
+            if (MMInitLayer(&hMiraMonLayerARC, pszFilename, nMMVersion,
+                            nMMRecode, nMMLanguage, nMMMemoryRatio, nullptr,
+                            MM_WRITTING_MODE, MMMap))
+            {
+                bValidFile = false;
+                return;
+            }
+            hMiraMonLayerARC.bIsBeenInit = 0;
+
+            CPLDebug("MiraMon", "Initializing MiraMon polygons layer...");
+            if (MMInitLayer(&hMiraMonLayerPOL, pszFilename, nMMVersion,
+                            nMMRecode, nMMLanguage, nMMMemoryRatio, nullptr,
+                            MM_WRITTING_MODE, MMMap))
+            {
+                bValidFile = false;
+                return;
+            }
+            hMiraMonLayerPOL.bIsBeenInit = 0;
+
+            // Just in case that there is no geometry but some other
+            // information to get. A DBF will be generated
+            CPLDebug("MiraMon", "Initializing MiraMon only-ext-DBF layer...");
+            if (MMInitLayer(&hMiraMonLayerReadOrNonGeom, pszFilename,
+                            nMMVersion, nMMRecode, nMMLanguage, nMMMemoryRatio,
+                            nullptr, MM_WRITTING_MODE, nullptr))
+            {
+                bValidFile = false;
+                return;
+            }
+            hMiraMonLayerPOL.bIsBeenInit = 0;
+
+            // This helps the map to be created
+            //GetLayerDefn()->SetName(hMiraMonLayerPNT.pszSrcLayerName);
+            poFeatureDefn->SetName(hMiraMonLayerPNT.pszSrcLayerName);
+        }
+
+        // Saving the HRS in the layer structure
+        if (poSRS)
+        {
+            if (poSRS->GetAuthorityName(nullptr) &&
```

Comment:
```suggestion
            const char* pszAuthorityName = poSRS->GetAuthorityName(nullptr);
            const char* pszAuthorityCode = poSRS->GetAuthorityCode(nullptr);
            if (pszAuthorityName && pszAuthorityCode &&
```
and replace below calls to GetAuthorityName/Code by the above variables

Reply:
Done